content/learn/beginner/datatypes-1.md

```diff
@@ -0,0 +1,131 @@
++++
+title = "Data Types 1"
+weight = 220
++++
+
+In this chapter, we will go over the basic data types that are available in the bot's template actions. These data types
+are used to store various types of data, such as numbers, strings, and more. We will also go over how to use variables
+to temporarily store and manipulate data.
+
+If you're completely new to programming, you might not know what a data type is. You can think of a data type as a way
+to distinguish between different kinds of "things". As a real-life analogy, you can separate food into several
+categories, such as fruits, vegetables, and meat. Each category is in that sense its own "data type".
+
+In programming, we have similar categories, such as numbers, strings, and booleans (true / false values).
+Each of these categories has its own set of operations that can be performed on them. For instance, you can add two
+numbers together, but you cannot add two strings together (at least not in the way you might expect).
+
+## Strings
+
+A string is a sequence of characters. You can think of it as a word or a sentence. In the bot's template actions, you
+can create a string by enclosing some text in double quotes (`"`). For instance, `"Hello, world!"` is a string. We call
+those *quoted strings*.
+
+Now, here we might run into a problem quite quickly: what if we want to include a double quote in our string? We can't
+just write `"Peter said "Hello, world!""`, as the bot would interpret this in a nonsensical way. To solve this, we
+must escape the double quote by adding a backslash (`\`) in front of it. This tells the bot that the double quote is
+part of the string and not the end of it. In other words, `"Peter said \"Hello, world!\""` would yield the expected
+result.
+
+To insert a newline (you would press `Enter` on your keyboard), you can use the escape sequence `\n`. Consider the
+string `"Hello\nWorld!"`, which would result in the following output:
+
+```txt
+Hello
+World!
+```
+
+For a full list of escape sequences, you can refer to the [Go documentation](https://golang.org/ref/spec#Rune_literals).
+Please note that not all escape sequences are supported by Discord.
+
+### Raw String Literals
+
+It should become relatively clear that a lot of new lines and other special characters can make a quoted string quite
+hard to read. To make this easier, you can use backticks (`` ` ``) to create a *raw string literal*. This enables you to
+*literally* press enter to insert a new line in your string, without having to use the `\n` escape sequence, like so:
```

Comment:
"A raw string literal does not attempt to interpret its contents in any way, and will simply contain the text between the opening \` and the closing \` unmodified."